File: contact_book/models/contact_book_history.py
# -*- coding: utf-8 -*-
##############################################################################
#
#    Copyright Domiup (<http://domiup.com>).
#
##############################################################################

from odoo import api, fields, models, _


class ContactBookHistory(models.Model):
    _name = 'contact.book.history'
    _description = 'History'
    _order = 'date desc'

    date = fields.Datetime('Date', required=True, default=fields.Datetime.now)
    type = fields.Selection([('Call', 'Call'), ('SMS', 'SMS')],
                           string='Type')
    user_id = fields.Many2one(
        comodel_name='res.users', default=lambda self: self.env.uid)
    contact_name = fields.Char('Contact')
    phone = fields.Char('Phone')
    email = fields.Char('Email')
    fa_icon = fields.Char('Fa-icon')
    res_id = fields.Integer('Record ID')
    res_model = fields.Char('Record Model')

File: contact_book/static/src/js/widgets/contact_book_history.js
odoo.define('contact_book.ContactBookHistory', function (require) {
"use strict";

var time = require('web.time');
var BasicWidget = require('contact_book.BasicWidget');


var ContactBookHistory = BasicWidget.extend({
    events:{
    },
    
    /**
     * @constructor
     */
    init: function (parent, options) {
        this._super.apply(this, arguments);
        this.model = 'contact.book.history';
        this.id = options.id;
        this.date = options.date;
        this.type = options.type;
        this.user_id = options.user_id;
        this.contact_name = options.contact_name;
        this.phone = options.phone;
        this.fa_icon = options.fa_icon;
        this.res_id = options.res_id;
        this.res_model = options.res_model;
        this.date_display = this.date ? moment(time.str_to_datetime(this.date)) : null;
    },
    /**
     * @override
     */
    

});

return ContactBookHistory;
});